Issue/#4/AddIndexSupport

Amila17 opened the pull request:
Support for Adding Indexes (Unique and Compound)
Updated TravelAgent class to HandleIndexes for the build script.

Commit:
Issue/#4/AddIndexSupport

Support for Adding Indexes (Unique and Compound)
Updated TravelAgent class to HandleIndexes for the build script.

feanz commented:
This should also deal with this I guess https://github.com/feanz/lhm.net/issues/1

Commit:
Issue/#4/AddIndexSupport

Added a new enum to hold the index order.
Added a new model to hold the index definition.
Updated the Migrator code to have an private method to build the query and public method to get the input.
Updated the Program class.

Commit:
Updated connection string

Commit:
Updated conn string

feanz commented on src/lhm.net/Migrator.cs:
```diff
@@ -50,6 +50,28 @@ public void RenameColumn(string oldColumnName, string newColumnName)
+            InternalAddIndex(indexName, isUnique, String.Join(", ", indexInfo.Select(ii => ii.ToString())));
+        }
+
+        private void InternalAddIndex(string indexName, bool isUnique, string columnDefinition)
```
Can this not just be an overload of AddIndex?

Amila17 replied:
Nice a private overload. Forgot about that. :)

feanz commented on src/lhm.net/Migrator.cs:
```diff
@@ -50,6 +50,28 @@ public void RenameColumn(string oldColumnName, string newColumnName)
+            InternalAddIndex(indexName, isUnique, indexDef.ToString());
+        }
+
+        public void AddCompoundIndex(string indexName, bool isUnique, params IndexDef[] indexInfo)
```
Can this not just be an overload of AddIndex

Amila17 replied:
Done

Commit:
Addressed comments

feanz commented:
:shipit:
